src/services/flags.js

```diff
@@ -1,7 +1,67 @@
 import fetch from 'node-fetch';
 import AbortController from 'abort-controller';
 
-const getFlags = async (host, sdkKey, httpTimeout, lastUpdatedTime, logger) => {
+const getFlags = async (sync0Complete, s3BucketHost, host, sdkKey, httpTimeout, lastUpdatedTime, logger) => {
+   
+    if(!sync0Complete){
+        return sync0FromS3(s3BucketHost, host, sdkKey, httpTimeout, lastUpdatedTime, logger)
+    } else{
+        return regularServerSync(host, sdkKey, httpTimeout, lastUpdatedTime, logger);        
+    }
+}
+
+
+const sync0FromS3 = async(s3BucketHost, host, sdkKey, httpTimeout, lastUpdatedTime, logger) => {
+    logger.info(`'GET': ${s3BucketHost}/${sdkKey}`);
+
+    const controller = new AbortController();
+
+    const timeout = setTimeout(() => {
+        controller.abort();
+    }, httpTimeout);
+
+    try {
+
+        let res = await fetch(`${s3BucketHost}/${sdkKey}`, {
+            method: 'GET',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            signal: controller.signal
+        })
+
+        if (res.status == 200) {
+            const body = await res.json();
+            if(body.data.flags.length){
+                return {
+                    "flags": body.data.flags,
+                    "lastUpdatedTime": null,
+                };
+            }else{
+                return regularServerSync( host, sdkKey, httpTimeout, lastUpdatedTime, logger);          
```

Comment:
It is best to have this method not know about regularSync. The getFlags method should decide which method to call. 

You should return true, only if the file was found and downloaded, false otherwise.

Reply:
Sure, I'll make this change.